seqr/views/apis/individual_api.py

```diff
@@ -706,9 +707,12 @@ def _get_record_updates(record, individual, invalid_values, allowed_assigned_ana
                 parsed_val = v
             else:
                 parsed_val = INDIVIDUAL_METADATA_FIELDS[k](v)
-                if (k not in {FEATURES_COL, ABSENT_FEATURES_COL} and parsed_val == getattr(individual, k)) or has_same_features:
+                if (
+                    (has_same_features and k in {FEATURES_COL, ABSENT_FEATURES_COL})
+                    or parsed_val == getattr(individual, k)
+                ):
                     parsed_val = None
-            if parsed_val:
+            if isinstance(parsed_val, bool) or parsed_val:
```

Comment:
Actioned, noting I've changed to `is False` to stop python type coercion, eg: `0 == False # True`.

Also noting that setting a field to `0`, or attempting to override with an empty string will not cause an update.

Reply:
oh thats a good point. Maybe instead of settling `parsed_val = None` when the value didn't change and then checking if `parsed_val` is defined, we should just check directly if the value has changed

Reply:
and then we should update that unit test to also set one of the values to 0 and check for that

Reply:
Couple suggestions based on what you want to check:

1. Always write changes, even if overwriting to an empty value. I don't mind this, because it gives a user to clear values, and at least in our case, collaborators usually just change the original metadata sheet to update.

```python
_parsed_val = INDIVIDUAL_METADATA_FIELDS[k](v)
if k in {FEATURES_COL, ABSENT_FEATURES_COL} and not has_same_features:
    update_record[k] = _parsed_val
elif _parsed_val != getattr(individual, k):
    update_record[k] = _parsed_val
```

2. Only write if changes + has value:

```python
_parsed_val = INDIVIDUAL_METADATA_FIELDS[k](v)
if (
    k in {FEATURES_COL, ABSENT_FEATURES_COL}
    and not has_same_features
    and _parsed_val
):
    update_record[k] = _parsed_val
elif (
    _parsed_val != getattr(individual, k)
    and (_parsed_val or _parsed_val in (0, False))
):
    update_record[k] = _parsed_val
```

Reply:
I like 1 better, I think it is cleaner. It does mean we might be inadvertently changing some values from None to empty string or vice versa but I think thats okay. Although I do think it can be done as 1 if statement instead of an if/elif

Reply:
I've applied (1) and combined the conditionals.